pkg/handlers/embedded_cluster_node_join_command.go

```diff
@@ -178,13 +169,37 @@ func (h *Handler) GetEmbeddedClusterNodeJoinCommand(w http.ResponseWriter, r *ht
 		return
 	}
 
-	JSON(w, http.StatusOK, GetEmbeddedClusterNodeJoinCommandResponse{
-		ClusterID:              install.Spec.ClusterID,
+	clusterUUID, err := uuid.Parse(install.Spec.ClusterID)
+	if err != nil {
+		logger.Error(fmt.Errorf("failed to parse cluster id: %w", err))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	var currentAppVersionLabel string
+	// attempt to get the current app version label from the installed app
+	installedApps, err := store.GetStore().ListInstalledApps()
+	if err == nil && len(installedApps) > 0 {
+		appVersion, err := store.GetStore().GetAppVersion(installedApps[0].ID, installedApps[0].CurrentSequence)
+		if err != nil {
+			logger.Error(fmt.Errorf("failed to get app version: %w", err))
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		currentAppVersionLabel = appVersion.VersionLabel
+	} else {
+		// if there are no installed apps, we can't get the current app version label
+		logger.Info("no installed apps found")
```

Comment:
i think this should fail. same if the list of installed apps != 1

Reply:
You can join nodes before installing an app - or at least that's my understanding of the sequencing

Reply:
the app would be installed at this point. the meaning of `installed` here is a bit different... it's not that the app got deployed, it means that the license (and airgap bundle) have been processed.

Reply:
Gotcha
I'd still rather fall back to "we don't include the app version in the response" instead of "you can't join nodes" though - this is an extra layer of safety check, but far from the only one